app/utils/currencyConversion.ts

```diff
@@ -64,19 +64,33 @@ export const paymentAmountToText = (
   locale = "en-US",
 ): string => {
   if (paymentAmount.currency === WalletCurrency.USD) {
-    return (paymentAmount.amount / 100).toLocaleString(locale, {
-      style: "decimal",
-      maximumFractionDigits: 2,
-      minimumFractionDigits: 2,
-    })
+    try {
+      const amount = paymentAmount.amount / 100
+      if (amount) {
+        return amount.toLocaleString(locale, {
+          style: "decimal",
+          maximumFractionDigits: 2,
+          minimumFractionDigits: 2,
+        })
+      }
+      throw Error("unable to parse USD amount to text")
+    } catch (error) {
+      console.error(error)
```

Comment:
Can we record the error in crashlytics as well.

Reply:
> Can we record the error in crashlytics as well.

Not sure if it's okay to import the crashlytics callback in this file. If that's fine, then I can add it.